src/peergos/shared/storage/ContentAddressedStorage.java

```diff
@@ -179,18 +201,47 @@ private static String encode(String component) {
         }
 
         @Override
-        public CompletableFuture<List<Multihash>> put(PublicKeyHash owner, PublicKeyHash writer, List<byte[]> signatures, List<byte[]> blocks) {
-            return put(owner, writer, signatures, blocks, "cbor");
+        public CompletableFuture<TransactionId> startTransaction(PublicKeyHash owner) {
+            if (! isPeergosServer) // TODO remove once IPFS implements the transaction api
+                return CompletableFuture.completedFuture(new TransactionId(Long.toString(r.nextInt(Integer.MAX_VALUE))));
+            return poster.get(apiPrefix + "transaction/start" + "?owner=" + encode(owner.toString()))
+                    .thenApply(raw -> new TransactionId(new String(raw)));
+        }
+
+        @Override
+        public CompletableFuture<Boolean> closeTransaction(PublicKeyHash owner, TransactionId tid) {
+            if (! isPeergosServer) // TODO remove once IPFS implements the transaction api
+                return CompletableFuture.completedFuture(true);
+            return poster.get(apiPrefix + "transaction/close?arg=" + tid.toString() + "&owner=" + encode(owner.toString()))
```

Comment:
perhaps extract actions into class constants. not so bothered by params

Reply:
done